### PR TITLE
Support crane

### PR DESCRIPTION
**Pull request by navyau09**

### CODE REVIEW

File: hooks/pre-command [hunk 1]
```diff
@@ -17,12 +17,23 @@ if docker --version >/dev/null 2>&1; then
...
+  echo "Logging in to crane as $USERNAME to $HOSTNAME"
+  crane auth login \
+    --username "$USERNAME" \
+    --password "$PASSWORD" \
+    "$HOSTNAME"
+
+else
+  echo "No cli is available to auth. Save creds to /.docker/config.json"
+  mkdir ~/.docker && touch ~/.docker/config.json
+  echo "$USERNAME:$PASSWORD" >~/.docker/config.json
```
Comment (alpar-t) [inline hooks/pre-command]: There's more to it, something like 
```
cat > ~/.docker/config.json <<EOF 
{
    "auths": {
        "$HOSTNAME": {
            "auth": "$(echo "$USERNAME:$PASSWORD" | base64)"
        }
    }
}
EOF
```

**Review by alpar-t — COMMENTED** [review]
to deploy this change, push a new git tag to this repo, then update pipeline to use the new tag.

File: hooks/pre-command [hunk 2]
```diff
@@ -17,12 +17,23 @@ if docker --version >/dev/null 2>&1; then
...
+  echo "Logging in to crane as $USERNAME to $HOSTNAME"
+  crane auth login \
+    --username "$USERNAME" \
+    --password "$PASSWORD" \
+    "$HOSTNAME"
+
+else
+  echo "No cli is available to auth. Save creds to /.docker/config.json"
+  mkdir ~/.docker && touch ~/.docker/config.json
+  echo "{auths: {$HOSTNAME: {auth: $(echo "$USERNAME:$PASSWORD" | base64)}}}" >~/.docker/config.json
```
Comment (alpar-t) [inline hooks/pre-command]: Have you tested this ? Are you sure it works ? Doesn't seem like valid json because of the missing quotes

Comment (navyau09) [reply hooks/pre-command]: I have tested now and fixed the syntax :)

**Review by alpar-t — APPROVED** [review]

**Pull request merged**